x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -127,6 +166,11 @@ export class TaskRunnerFactory {
           logger.error(
             `Failed to cleanup action_task_params object [id="${actionTaskParamsId}"]: ${e.message}`
           );
+        } finally {
+          // if subApiKey exists, it should be invalidated after usage
+          if (subApiKeyId) {
+            await invalidateAPIKey({ id: subApiKeyId }, securityPluginSetup!);
+          }
```

Comment:
We can remove this from `finally` and wrap it within a condition of `executorResult.status === 'success'`.